frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDrawerRightTabs.tsx

```diff
@@ -35,47 +24,64 @@ const PipelineRunDrawerRightTabs: React.FC<PipelineRunDrawerRightTabsProps> = ({
+  const hasNoInputsOutputs = !task.inputs && !task.outputs;
+  const [selection, setSelection] = React.useState<string>(
+    hasNoInputsOutputs ? PipelineRunNodeTab.DETAILS : PipelineRunNodeTab.INPUT_OUTPUT,
+  );
   const taskExecution = executions.find(
     (e) => e.getCustomPropertiesMap().get('task_name')?.getStringValue() === task.name,
   );
 
-  const tabContents: Record<PipelineRunNodeTabs, React.ReactNode> = {
-    [PipelineRunNodeTabs.INPUT_OUTPUT]: (
-      <SelectedNodeInputOutputTab task={task} execution={taskExecution?.toObject()} />
-    ),
-    // [PipelineRunNodeTabs.VISUALIZATIONS]: <>TBD 2</>,
-    [PipelineRunNodeTabs.DETAILS]: <SelectedNodeDetailsTab task={task} />,
-    [PipelineRunNodeTabs.VOLUMES]: <SelectedNodeVolumeMountsTab task={task} />,
-    [PipelineRunNodeTabs.LOGS]: <LogsTab task={task} />,
-    // [PipelineRunNodeTabs.POD]: <>TBD 6</>,
-    // [PipelineRunNodeTabs.EVENTS]: <>TBD 7</>,
-    // [PipelineRunNodeTabs.ML_METADATA]: <>TBD 8</>,
+  const tabs: Record<string, { title: string; isDisabled?: boolean; content: React.ReactNode }> = {
+    [PipelineRunNodeTab.INPUT_OUTPUT]: {
+      title: 'Input/Output',
+      isDisabled: hasNoInputsOutputs,
+      content: <SelectedNodeInputOutputTab task={task} execution={taskExecution?.toObject()} />,
+    },
+    [PipelineRunNodeTab.DETAILS]: {
+      title: 'Task details',
+      content: <SelectedNodeDetailsTab task={task} />,
+    },
+    [PipelineRunNodeTab.VOLUMES]: {
+      title: 'Volumes',
+      isDisabled: !task.volumeMounts || task.volumeMounts.length === 0,
+      content: renderDetailItems(
+        (task.volumeMounts ?? []).map(({ name, mountPath }) => ({ key: mountPath, value: name })),
+      ),
+    },
+    [PipelineRunNodeTab.LOGS]: {
+      title: 'Logs',
+      isDisabled: !task.status?.podName,
+      content: <LogsTab task={task} />,
+    },
   };
 
   return (
     <>
       <Tabs activeKey={selection} mountOnEnter>
-        {Object.values(PipelineRunNodeTabs).map((tab) => (
+        {Object.entries(tabs).map(([tabName, { title, isDisabled }]) => (
           <Tab
-            data-testid={`right-drawer-tab-${tab}`}
-            key={tab}
-            title={PipelineRunNodeTabsTitles[tab]}
-            eventKey={tab}
-            tabContentId={tab}
-            onClick={() => setSelection(tab)}
+            data-testid={`right-drawer-tab-${tabName}`}
+            key={tabName}
+            title={title}
+            eventKey={tabName}
+            tabContentId={tabName}
+            isDisabled={isDisabled}
+            onClick={() => setSelection(tabName)}
           />
         ))}
       </Tabs>
       <DrawerPanelBody className="pipeline-run__drawer-panel-body pf-v5-u-px-sm">
-        <TabContent
-          id={selection}
-          eventKey={selection}
-          activeKey={selection}
-          style={{ flex: '1 1 auto' }}
-        >
-          {tabContents[selection]}
-        </TabContent>
+        {!tabs[selection].isDisabled && (
```

Comment:
is this needed? `!tabs[selection].isDisabled`

Reply:
I wasn't sure if there was a scenario where maybe some isDisabled condition still leading to one of the default tabs (Summary or Inputs/Outputs) being displayed. isDisabled doesn't prevent the rendering of the content within the tab. If you want me to take it out considering this, let me know.

Reply:
doesnt really affect the logic, we can keep it